cpp/src/sampling/detail/graph_functions.hpp

```diff
@@ -30,48 +30,67 @@ namespace detail {
  *
  * Collect all the edges that are present in the adjacency lists on the current gpu
  *
- * @tparam GraphViewType Type of the passed non-owning graph object.
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @tparam edge_t Type of edge identifiers. Needs to be an integral type.
+ * @tparam weight_t Type of edge weights. Needs to be a floating point type.
+ * @tparam edge_type_t Type of edge type. Needs to be an integral type.
+ * @tparam label_t Type of label. Needs to be an integral type.
+ * @tparam multi_gpu Flag indicating whether template instantiation should target single-GPU (false)
+ *
  * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
  * handles to various CUDA libraries) to run graph algorithms.
- * @param graph_view Non-owning graph object.
+ * @param graph_view Graph View object to generate NBR Sampling on.
+ * @param edge_weight_view Optional view object holding edge weights for @p graph_view.
+ * @param edge_id_view Optional view object holding edge ids for @p graph_view.
+ * @param edge_type_view Optional view object holding edge types for @p graph_view.
  * @param active_majors Device vector containing all the vertex id that are processed by
  * gpus in the column communicator
+ * @param active_major_labels Optional device vector containing labels for each device vector
```

Comment:
Do we ever use this without valid `active_major_labels`? (If it is invalid, isn't this just decompressing a graph to edge list?).